src/test/regress/sql/subquery_basics.sql

```diff
@@ -360,3 +360,19 @@ FROM
 		ORDER BY 1
 		LIMIT 5
 	) as foo WHERE user_id IN (SELECT count(*) FROM users_table GROUP BY user_id);
+
+CREATE TABLE dist(id int);
+SELECT create_distributed_table('dist','id');
+CREATE TABLE ref(id int);
+SELECT create_reference_table('ref');
+
+-- the outer join checks converts SELECT .. FROM ref FULL JOIN intermediate_result WHERE .. (sublink)
+SELECT * FROM ref FULL JOIN dist USING (id)
+WHERE
+	dist.id IN (SELECT id FROM dist GROUP BY id);
+
+-- sublinks in the targetlist are not supported
+SELECT (SELECT id FROM dist WHERE dist.id > d1.id GROUP BY id) FROM ref FULL JOIN dist d1 USING (id);
```

Comment:
I think we can expand this test such that it fails on the `main` branch due to not matching target lists of the intermediate results:

```SQL
CREATE TABLE dist_1(id int, value text);
SELECT create_distributed_table('dist_1','id');
CREATE TABLE ref_1(id int);
SELECT create_reference_table('ref_1');

INSERT INTO dist_1 VALUES (1, '1');
INSERT INTO ref_1 VALUES (1);


SELECT * FROM ref_1 FULL JOIN dist_1 USING (id)
WHERE
	dist_1.id IN (SELECT id FROM dist_1 GROUP BY id);

ERROR:  row field count is 2, expected 1
CONTEXT:  while executing command on localhost:5432
```